docs/4.2. Behaviour - Clients.md

```diff
@@ -20,22 +20,42 @@ containing the required details).
 
 Clients are RECOMMENDED to provide for the OAuth 2.0 Dynamic Client Registration Protocol [RFC 7591][RFC-7591] to allow
 for zero-configuration setup. NMOS Nodes SHOULD be capable of registering dynamically in order to limit the
-configuration associated with large numbers of Nodes coming online simultaneously. Clients MUST include a populated
-`scope` parameter when performing dynamic registrations.
+configuration associated with large numbers of Nodes coming online simultaneously. Clients MUST include populated
+`client_name` and `scope` parameters when performing dynamic registrations. The `client_name` MUST be unique to each
+instance of the software or hardware. It is RECOMMENDED that the `client_name` include the manufacturer's name, product
+identifier and serial number.
 
 Clients that do not register with an explicit `token_endpoint_auth_method` parameter will be registered as a
 Confidential Client and issued with a client secret. Public clients MUST specify a value of "none" for this parameter
 to notify the Authorization Server that is is a public client.
 
+Clients which require use of the `client_credentials` grant type MUST support the following means of authentication
+during the dynamic client registration process. All other Clients SHOULD support these authentication methods:
+*   Presentation of an initial JSON Web Token (JWT) as suggested in Section 3 of [RFC 7591][RFC-7591]
+*   Unauthenticated registration
+
+Clients MUST be configurable to choose between the above authentication methods, which are listed in order of
+preference. When an initial JWT authentication method is selected, Clients MUST provide the means to configure a custom
+value for this initial JWT.
+
 Clients MAY check and modify existing registrations with an Authorization Server as-per [RFC 7592][RFC-7592]. As this
 is an optional feature of Authorization Servers, Clients MUST confirm that this is implemented as indicated by the
 presence of the keys `registration_client_uri` and `registration_access_token` in a client registration response.
 
 ## Client Credentials
 
-Clients operating with a client secret SHOULD support using HTTP Basic Authentication, as per Section 2 of
-[RFC 2617][RFC-2617], to authenticate with the Authorization Server in the manner described in Section 2.3.1 of
-[RFC 6749][RFC-6749].
+Confidential Clients SHOULD support authentication using a digitally signed JSON Web Token, to authenticate with the
```

Comment:
It might be best to highlight to vendors the need for clients to be able to support hosting their own public keys for validation of JWTs used for authentication. Something along the lines of:
```Clients SHOULD be capable of hosting their own public JWKs at a location that corresponds to the value of the registered jwks_uri URL used when registering with the Auth Server, if it was supplied```

Reply:
Also should memtion `jwks_uri` is preferred over the `jwks`, allowing easier key rotation.